### DISCUSSION TITLE
macOS file access Bookmark can't be restored

### DISCUSSION THREAD

**Discussion by pzs7602** [AvaloniaUI/Avalonia#14852] [Q&A] [answered]
In my avalonia 11.0.9 project(macOS sandbox app), I use StorageProvider method OpenFolderPickerAsync to ask user to choose a folder, then save its bookmark for later usage:
var BookmarkID = await folder.SaveBookmarkAsync();
the returned bookmarkid is saved to a file
next time when the app starts again, it read the bookmark file and try to restore the bookmark:
MainWinObj.StorageProvider.OpenFolderBookmarkAsync(bk.Id)
but accessing the files inside the folder are still denied. Is the bookmark in macOS can't be persisted?

**Comment by maxkatz6** [comment, marked as answer]
SaveBookmarkAsync wasn't yet properly implemented on macOS.
We need to reopen https://github.com/AvaloniaUI/Avalonia/pull/6540 PR and implement it on top of new APIs.
If somebody can try this task - it will be helpful.
